bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/api/BErrorCreator.java

```diff
@@ -60,7 +62,12 @@ public static BError createError(BString message, BString details) {
         if (details != null) {
             detailMap.put(ERROR_MESSAGE_FIELD, details);
         }
-        return new ErrorValue(message, detailMap);
+        return new ErrorValue(message, readonlyCloneDetailMap(detailMap));
```

Comment:
This way we're creating two maps, first the mutable one and then the immutable one.

Instead, shall we use https://github.com/ballerina-platform/ballerina-lang/blob/master/bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/MapValueImpl.java#L101?

We can use https://github.com/ballerina-platform/ballerina-lang/blob/master/bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ReadOnlyUtils.java#L72 to get the immutable type and then pass the `detail` as a `KeyValueEntry` https://github.com/ballerina-platform/ballerina-lang/blob/master/bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/MappingInitialValueEntry.java#L39.

Reply:
The other helper methods here should also be updated, right? They don't seem to be making the error detail immutable? 

IMO, we should also check on the possibility of making `org.ballerinalang.jvm.types.BTypes#typeErrorDetail` itself be immutable (i.e., setting the `readonly` flag). @rdhananjaya, WDYT?

Reply:
Agreed `org.ballerinalang.jvm.types.BTypes#typeErrorDetail` should also be the readonly varient.